org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -888,6 +941,25 @@ public Preferences setJavaCompletionFavoriteMembers(List<String> javaCompletionF
 		return this;
 	}
 
+	public Preferences putSha256(List<String> sha256Allowed, List<String> sha256Disallowed) {
+		List<String> oldAllowed = this.sha256Allowed;
+		List<String> oldDisallowed = this.sha256Disallowed;
+		WrapperValidator.getAllowed().clear();
+		this.sha256Allowed = sha256Allowed;
+		WrapperValidator.getDisallowed().clear();
+		if (sha256Disallowed != null) {
+			WrapperValidator.getDisallowed().addAll(sha256Disallowed);
+		}
+		this.sha256Disallowed = sha256Disallowed;
+		ProjectsManager projectsManager = JavaLanguageServerPlugin.getProjectsManager();
+		if (projectsManager != null && (!Objects.equals(oldAllowed, this.sha256Allowed) || !Objects.equals(oldDisallowed, this.sha256Disallowed))) {
+			for (IProject project : ProjectUtils.getGradleProjects()) {
```

Comment:
I don't like performing stuff like that here in the setter. For later: use a preference listener to do that kind of things